subprojects/docs/src/docs/userguide/migration/upgrading_version_8.adoc

```diff
@@ -40,6 +40,56 @@ The previous step will help you identify potential problems by issuing deprecati
 
 === Potential breaking changes
 
+==== ArtifactView and ArtifactCollection variant resolution behavior is more correct
+
+In previous versions of Gradle, the attributes used for variant resolution by `ArtifactView` and `ArtifactCollection` began as point-in-time copies of the attributes present on the `Configuration` instance used to create them.
+
+This meant the attributes used for resolution by those types could unexpectedly differ when the attributes on the `Configuration` changed after an `ArtifactView` or `ArtifactCollection` was created.
+Some default attributes are lazily and automatically added to `Configuration` instances by core Java plugins, so could lead to unexpected behavior.
+
+Consider the following scenario where `producer` is a sibling `java-library` project to the build below:
+
+```groovy
+    plugins {
+        id 'java-library'
+    }
+
+    dependencies {
+        api project(':producer')
+    }
+
+    def incomingFiles = configurations.compileClasspath.incoming.files
+    def artifactViewFiles = configurations.compileClasspath.incoming.artifactView { }.files
+
+    logger.warn 'Incoming Files:'
+    incomingFiles.each { logger.warn 'Name: ' + it.name }
+
+    logger.warn 'Artifact View Files:'
+    artifactViewFiles.each { logger.warn 'Name: ' + it.name }
```

Comment:
```suggestion
    println 'Incoming Files:'
    incomingFiles.each { println 'Name: ' + it.name }

    println 'Artifact View Files:'
    artifactViewFiles.each { println 'Name: ' + it.name }
```

I think in most places we traditionally use println instead of logger.warn. Any objection to changing this for the sake of consistency? Applies to the tests as well.

Reply:
No objection, I'll update it.

Reply:
This looks like it was not updated here?